module/VuFindSearch/src/VuFindSearch/Backend/EDS/Backend.php

```diff
@@ -295,30 +295,47 @@ public function retrieve($id, ParamBag $params = null)
                 $this->profile = $overrideProfile;
             }
             $sessionToken = $this->getSessionToken();
-            $parts = explode(',', $id, 2);
-            if (!isset($parts[1])) {
-                throw new BackendException(
-                    'Retrieval id is not in the correct format.'
+
+            $backendType = (null !== $params) ? $params->get('backendType')[0] : null;
```

Comment:
@demiankatz I do not like this $backendType, but did it anyway because it's tentatively the only significant change to the EDS Backend, and if that stays the same after all the MVP is done, it may be a better option than a second backend.  But I'm not convinced yet.  The two APIs share so much in common, despite being defined separately; it's very confusing.

Reply:
Rather than passing this in through the ParamBag, would it make more sense to pass it in through the constructor, and then address it only at the factory level? I assume it's going to be constant for each backend, and we'll have a different instance of the class for each backend, so I don't think there's a reason to pass the value dynamically. And, of course, if it eventually turns out to make more sense to have separate subclasses for each service, you still only have to change the factory code to make the adjustment, instead of potentially having to eliminate some extra ParamBag-based communication.

Reply:
Setting the backendType via the backend factory instead.  Could be done in the constructor but I used a mutator instead as that seemed to be the pattern in that file.

Reply:
Works for me -- thanks!